scipy/stats/tests/test_qmc.py

```diff
@@ -0,0 +1,872 @@
+        base_engine = qmc.Sobol(d=2, scramble=False, seed=seed)
+        engine = qmc.MultivariateNormalQMC(mean=np.zeros(2),
+                                           engine=base_engine,
+                                           inv_transform=True, seed=seed)
+        samples = engine.random()
+        assert_equal(samples.shape, (1, 2))
+
+    def test_NormalQMCSeeded(self):
+        # test even dimension
+        seed = np.random.RandomState(12345)
+        engine = qmc.MultivariateNormalQMC(
+            mean=np.zeros(2), inv_transform=False, seed=seed)
+        samples = engine.random(n=2)
+        samples_expected = np.array(
+            [[-0.943472, 0.405116], [-0.63099602, -1.32950772]]
+        )
+        assert_array_almost_equal(samples, samples_expected)
+
+        # test odd dimension
+        seed = np.random.RandomState(12345)
+        engine = qmc.MultivariateNormalQMC(
+            mean=np.zeros(3), inv_transform=False, seed=seed)
+        samples = engine.random(n=2)
+        samples_expected = np.array(
+            [
+                [-0.943472, 0.405116, 0.268828],
+                [1.83169884, -1.40473647, 0.24334828],
+            ]
+        )
+        assert_array_almost_equal(samples, samples_expected)
+
+    def test_NormalQMCSeededInvTransform(self):
+        # test even dimension
+        seed = np.random.RandomState(12345)
+        engine = qmc.MultivariateNormalQMC(
+            mean=np.zeros(2), seed=seed, inv_transform=True)
+        samples = engine.random(n=2)
+        samples_expected = np.array(
+            [[0.228309, -0.162516], [-0.41622922, 0.46622792]]
+        )
+        assert_array_almost_equal(samples, samples_expected)
+
+        # test odd dimension
+        seed = np.random.RandomState(12345)
+        engine = qmc.MultivariateNormalQMC(
+            mean=np.zeros(3), seed=seed, inv_transform=True)
+        samples = engine.random(n=2)
+        samples_expected = np.array(
+            [
+                [0.228309, -0.162516, 0.167352],
+                [-1.40525266, 1.37652443, -0.8519666],
+            ]
+        )
+        assert_array_almost_equal(samples, samples_expected)
+
+    def test_NormalQMCShapiro(self):
+        seed = np.random.RandomState(12345)
+        engine = qmc.MultivariateNormalQMC(mean=np.zeros(2), seed=seed)
+        samples = engine.random(n=256)
+        assert_(all(np.abs(samples.mean(axis=0)) < 1e-2))
+        assert_(all(np.abs(samples.std(axis=0) - 1) < 1e-2))
+        # perform Shapiro-Wilk test for normality
+        for i in (0, 1):
+            _, pval = shapiro(samples[:, i])
+            assert_(pval > 0.9)
+        # make sure samples are uncorrelated
+        cov = np.cov(samples.transpose())
+        assert_(np.abs(cov[0, 1]) < 1e-2)
+
+    def test_NormalQMCShapiroInvTransform(self):
+        seed = np.random.RandomState(12345)
+        engine = qmc.MultivariateNormalQMC(
+            mean=np.zeros(2), seed=seed, inv_transform=True)
```

Comment:
Did the default for `inv_transform` used to be `False`? In any case, it is `True` now, so isn't this test identical to the previous one?

This brings to mind that there is a lot of duplication here. It is easier to review and maintain the test suite if we combine what we can and use `pytests.mark.parametrize` or subclasses of test classes to repeat tests with different options.